//
// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package scql.engine.csv;

message CsvTableConf {
  message ColumnConf {
    string column_name = 1;
    // column_type value should be string value defined in DataType in
    // "api/column.proto"
    string column_type = 2;
  };

  string table_name = 1;
  string data_path = 2;
  repeated ColumnConf columns = 3;
}

message S3Conf {
  string endpoint = 2;
  string access_key_id = 3;
  string secret_access_key = 4;
  // See
  // https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html
  bool virtualhost = 5;
}

message CsvdbConf {
  string db_name = 1;
  S3Conf s3_conf = 2;
  repeated CsvTableConf tables = 3;
}